ts/Maps/MapNavigation.ts

```diff
@@ -252,13 +256,65 @@ MapNavigation.prototype.update = function (
                             buttonOptions.alignTo
                         );
                     }
+
                     unbind();
                 });
             } else {
                 button.align(buttonOptions, false, buttonOptions.alignTo);
             }
-
         });
+
+        // Borrowed from overlapping-datalabels. Consider a shared module.
+        const isIntersectRect = (
+            box1: BBoxObject,
+            box2: BBoxObject
+        ): boolean => !(
+            box2.x >= box1.x + box1.width ||
+            box2.x + box2.width <= box1.x ||
+            box2.y >= box1.y + box1.height ||
+            box2.y + box2.height <= box1.y
+        );
+
+        // Check the mapNavigation buttons collision with exporting button
+        // and translate the mapNavigation button if they overlap.
+        const adjustMapNavBtn = function (): void {
+            const expBtnBBox =
+                    chart.exportingGroup && chart.exportingGroup.getBBox();
+
+            if (expBtnBBox) {
+                const navBtnsBBox = chart.navButtonsGroup.getBBox();
+
+                // If buttons overlap
+                if (isIntersectRect(expBtnBBox, navBtnsBBox)) {
+                    // Adjust the mapVan buttons' position by translating them
+                    // above or below the exporting button
+                    const aboveExpBtn = -navBtnsBBox.y - navBtnsBBox.height +
+                            expBtnBBox.y - 5,
+                        belowExpBtn = expBtnBBox.y + expBtnBBox.height -
+                            navBtnsBBox.y + 5,
+                        mapNavVerticalAlign =
+                            o.buttonOptions && o.buttonOptions.verticalAlign;
+
+                    // If bottom aligned and adjusting the mapNav button would
+                    // translate it out of the plotBox, translate it up
+                    // instead of down
+                    chart.navButtonsGroup.attr({
+                        translateY: mapNavVerticalAlign === 'bottom' ?
+                            aboveExpBtn :
+                            belowExpBtn
+                    });
+                }
+            }
+        };
+
+        if (!chart.hasLoaded) {
+            // Align it after the plotBox is known (#12776)
+            // and after the hamburger button's position is known
+            // so they don't overlap (#15782)
+            addEvent(chart, 'render', (): void => {
```

Comment:
The callback can be passed directly. `addEvent(chart, 'render', adjustMapNavBtn);`